[ROR 1.26.0][RORDEV-269] response_fields rule

pielas opened the pull request:
I see that atom removed unnecessary spaces from this document. Let me know if that's ok

coutoPL commented on elasticsearch.md:
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
 
+#### `response_fields`
+
+This rule allows filtering Elasticsearch responses using defined list of fields. It works in very similar way to fields rule. However it **doesn't make use of Field Level Security \(FLS\)** and could be applied to every response returned by Elasticsearch.
```
to the `fields` rule

coutoPL replied:
maybe we should mention that in contrast to `fields` rule which filters out document fields, this rule filters our response fields?

coutoPL commented on elasticsearch.md:
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+This rule allows filtering Elasticsearch responses using defined list of fields. It works in very similar way to fields rule. However it **doesn't make use of Field Level Security \(FLS\)** and could be applied to every response returned by Elasticsearch.
+
+It could be configured in two modes: *whitelist* passing only defined fields or *blacklist* filtering out defined fields.
```
maybe bullet list for the modes?

coutoPL commented on elasticsearch.md:
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+`fields: ["~excluded_fields_prefix_*", "~excluded_field", "~another_excluded_field.nested_field"]`
+
+Return documents but deprived of the fields that:
```
let's don't talk about document in this rule. This rule filters response fields, not document fields (like `fields` rule)

coutoPL commented on elasticsearch.md:
```diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+`response_fields: ["allowed_fields_prefix_*", "_*", "allowed_field.nested_field.text"]`
+
+Return documents deprived of all the fields, except the ones that:
```
same here

coutoPL commented on elasticsearch.md:
````diff
@@ -920,6 +920,46 @@ After adding the `filter` rule \(using the block duplication strategy\).
+  response_fields: ["cluster_name", "status"]
+```
+
+Any field could be filtered using this rule.
````
maybe we should show an example of full `/_cluster/health`response body and the filtered one?

coutoPL commented:
@pielas please set `develop` as a target branch
@sscarduzio please also review the PR and change default branch to develop

coutoPL reviewed (changes requested):
@pielas what has happened? I see whole file changed? There is really hard to figure out what has changed

pielas commented:
> @pielas what has happened? I see whole file changed? There is really hard to figure out what has changed

Atom somehow changed new line characters, this is why there was change in whole file. I fixed it and squashed my commits with review fixes

coutoPL reviewed (approved):
LGTM. @sscarduzio do you want to take a look?

sscarduzio commented on elasticsearch.md:
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
 
+#### `response_fields`
+
+This rule allows filtering Elasticsearch responses using defined list of fields. It works in very similar way to `fields` rule. However in contrast to `fields` rule, which filters out document fields, this rule filters out response fields. It **doesn't make use of Field Level Security \(FLS\)** and could be applied to every response returned by Elasticsearch.
```
"using a list of fields".

sscarduzio replied:
Remove that "However".

sscarduzio replied:
~could~  -> can

 (apply also below.)

sscarduzio commented on elasticsearch.md:
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+It could be configured in two modes:
+  * *whitelist* passing only defined fields
+  * *blacklist* filtering out defined fields.
```
filtering out (removing) only the defined fields from the response JSON object

pielas replied:
I removed "JSON" because this rule can also filter yaml or binary cbor responses

sscarduzio commented on elasticsearch.md:
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+**Blacklist mode**
+
+Specifies which fields should be filtered out with `~` (other fields from mapping become allowed implicitly). Example:
```
filtered out by adding the `~` prefix to the field name. Other fields in the response will be implicitly allowed. For example:

sscarduzio commented on elasticsearch.md:
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+`fields: ["~excluded_fields_prefix_*", "~excluded_field", "~another_excluded_field.nested_field"]`
+
+Return response but deprived of the fields that:
```
The above will return the usual response object, but deprived (if found) of the fields that:

sscarduzio commented on elasticsearch.md:
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+Specifies which fields should be filtered out with `~` (other fields from mapping become allowed implicitly). Example:
+
+`fields: ["~excluded_fields_prefix_*", "~excluded_field", "~another_excluded_field.nested_field"]`
```
"fields" -> "respone_fields"

sscarduzio commented on elasticsearch.md:
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
+
+**Whitelist mode**
+
+In this mode rule is configured to filter out each field that isn't defined in rule.
```
in THE rule

sscarduzio commented on elasticsearch.md:
````diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
+}
+```
+
+**NB:** Any field could be filtered using this rule.
````
Any response field can be filtered using this rule.

sscarduzio reviewed (changes requested):
Some changes for clarity + one typo

pielas commented on elasticsearch.md:
```diff
@@ -920,6 +920,76 @@ After adding the `filter` rule \(using the block duplication strategy\).
+This rule allows filtering Elasticsearch responses using defined list of fields. It works in very similar way to `fields` rule. However in contrast to `fields` rule, which filters out document fields, this rule filters out response fields. It **doesn't make use of Field Level Security \(FLS\)** and could be applied to every response returned by Elasticsearch.
+
+It could be configured in two modes:
+  * *whitelist* passing only defined fields
```
I removed "JSON" because this rule can also filter yaml or binary cbor responses
